File: src/ajax/updateTugas.php
<?php
require '../utilities/db.php';
require '../utilities/model.php';
require '../utilities/view.php';
require '../models/tugas.php';
require '../models/komentar.php';

session_set_cookie_params(30 * 24 * 60 * 60); 
session_start();


$data = Array();
try {
	if (isset($_GET["removec"]) && isset($_GET["id_komentar"])) {
		$comment = new Komentar();

		$datakomentar = $comment->getCommentById($_GET["id_komentar"]);
		if ($_SESSION["user"]["username"] == $datakomentar["user"])
		{
			$comment->delete($_GET["id_komentar"]);
			$tugas = new Tugas();
			$tugas->updateTimestamp($datakomentar["id_tugas"]);

			$data["responseStatus"] = 200;
		}
		else
		{
			$data["responseStatus"] = 403;
			$data["message"] = "Forbidden";
		}
	} else if (isset($_POST["addc"]) && isset($_POST["id_tugas"]) && isset($_POST["content"])) {
		$comment = new Komentar();

	    $comment->set_id_tugas($_POST["id_tugas"]);
	    $comment->set_username($_SESSION["user"]["username"]);
	    $comment->set_content($_POST["content"]);
	    $comment->store();

		$tugas = new Tugas();
		$tugas->updateTimestamp($_POST["id_tugas"]);

		$data["responseStatus"] = 200;
	} else if (isset($_GET["chstatus"]) && isset($_GET["id_tugas"]) && isset($_GET["status"])) {
		$tugas = new Tugas();
		$tugas->setStats($_GET["id_tugas"], $_GET["status"]);

		$data["responseStatus"] = 200;
	} else if (isset($_GET["removea"]) && isset($_GET["id_tugas"]) && isset($_GET["username"])) {
		$tugas = new Tugas();
		$tugas->removeAssignee($_GET["id_tugas"], $_GET["username"]);

		if ($_GET["username"] == $_SESSION["user"]["username"])
		{
			$data["killself"] = 1;
		}
		else
		{		
			$data["killself"] = 0;
		}
		
		$data["responseStatus"] = 200;
	} else if (isset($_GET["adda"]) && isset($_GET["id_tugas"]) && isset($_GET["username"])) {
		$tugas = new Tugas();
		$tugas->addAssignee($_GET["id_tugas"], $_GET["username"]);
		$tugas->updateTimestamp($_GET["id_tugas"]);

		$data["responseStatus"] = 200;
	} else if (isset($_GET["suggesta"]) && isset($_GET["id_tugas"]) && isset($_GET["start"])) {
		$tugas = new Tugas();
		$data = Array();
		$data["responseStatus"] = 200;
		$data["suggestedAssignees"] = $tugas->getSuggestionAssignees($_GET["id_tugas"], $_GET["start"] . '%', 10);
	} else {
		$data["responseStatus"] = 400;
		$data["message"] = "Bad request";
	}
} catch (Exception $e) {
	$data["responseStatus"] = 400;
	$data["message"] = $e->getMessage();
}

header('Content-Type:application/json');
echo json_encode($data);

File: TodolistJava/src/java/id/ac/itb/todolist/ajax/UpdateTugas.java
package id.ac.itb.todolist.ajax;

import id.ac.itb.todolist.dao.CommentDao;
import id.ac.itb.todolist.dao.TugasDao;
import id.ac.itb.todolist.json.JSONArray;
import id.ac.itb.todolist.json.JSONObject;
import id.ac.itb.todolist.model.Comment;
import id.ac.itb.todolist.model.User;
import java.io.IOException;
import java.io.PrintWriter;
import java.util.List;
import javax.servlet.ServletException;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import javax.servlet.http.HttpSession;

public class UpdateTugas extends HttpServlet {

    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        doGet(request, response);
    }

    @Override
    protected void doGet(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        response.setContentType("application/json");
        PrintWriter out = response.getWriter();
        HttpSession session = request.getSession();

        JSONObject jObject = new JSONObject();
        try {
            if (request.getParameter("removec") != null && request.getParameter("id_komentar") != null) {
                CommentDao commentDao = new CommentDao();
                int idComment = Integer.parseInt(request.getParameter("id_komentar"));

                Comment comment = commentDao.getCommentById(idComment);
                if (((User) session.getAttribute("user")).getUsername().equals(comment.getUser().getUsername())) {
                    commentDao.deleteComment(idComment);
                    TugasDao tugasDao = new TugasDao();
                    tugasDao.updateTimestamp(comment.getIdTugas());

                    jObject.put("responseStatus", 200);
                } else {
                    jObject.put("responseStatus", 403);
                    jObject.put("message", "Forbidden");
                }
            } else if (request.getParameter("addc") != null && request.getParameter("id_tugas") != null && request.getParameter("content") != null) {
                CommentDao commentDao = new CommentDao();
                Comment comment = new Comment();

                int idTugas = Integer.parseInt(request.getParameter("id_tugas"));
                String content = request.getParameter("content");

                comment.setIdTugas(idTugas);
                comment.setUser((User) session.getAttribute("user"));
                comment.setContent(content);
                commentDao.addComment(comment);

                TugasDao tugasDao = new TugasDao();
                tugasDao.updateTimestamp(idTugas);

                jObject.put("responseStatus", 200);
            } else if (request.getParameter("chstatus") != null && request.getParameter("id_tugas") != null && request.getParameter("status") != null) {
                TugasDao tugasDao = new TugasDao();
                
                int idTugas = Integer.parseInt(request.getParameter("id_tugas"));
                boolean status = Boolean.parseBoolean(request.getParameter("status"));
                
                tugasDao.setStatus(idTugas, status);

                jObject.put("responseStatus", 200);
            } else if (request.getParameter("updatedeadline") != null && request.getParameter("id_tugas") != null && request.getParameter("deadline") != null) {
                TugasDao tugasDao = new TugasDao();
                
                int idTugas = Integer.parseInt(request.getParameter("id_tugas"));
                java.sql.Date deadline = java.sql.Date.valueOf(request.getParameter("deadline"));
                
                tugasDao.setDeadline(idTugas, deadline);

                jObject.put("responseStatus", 200);
            } else if (request.getParameter("adda") != null && request.getParameter("id_tugas") != null && request.getParameter("username") != null) {
                TugasDao tugasDao = new TugasDao();
                
                int idTugas = Integer.parseInt(request.getParameter("id_tugas"));
                String username = request.getParameter("username");
                
                tugasDao.addAssignee(idTugas, username);
                tugasDao.updateTimestamp(idTugas);

                jObject.put("responseStatus", 200);
            } else if (request.getParameter("removea") != null && request.getParameter("id_tugas") != null && request.getParameter("username") != null) {
                TugasDao tugasDao = new TugasDao();
                
                int idTugas = Integer.parseInt(request.getParameter("id_tugas"));
                String username = request.getParameter("username");
                
                tugasDao.removeAssignee(idTugas, username);

                if (username.equals(((User) session.getAttribute("user")).getUsername())) {
                    jObject.put("killself", 1);
                } else {
                    jObject.put("killself", 0);
                }

                jObject.put("responseStatus", 200);
            } else if (request.getParameter("suggesta") != null && request.getParameter("id_tugas") != null && request.getParameter("start") != null) {
                TugasDao tugasDao = new TugasDao();
                
                int idTugas = Integer.parseInt(request.getParameter("id_tugas"));
                String startWith = request.getParameter("start") + "%";
                
                jObject.put("suggestedAssignees", new JSONArray(tugasDao.getSuggestionAssignees(idTugas, startWith, 10)));
                jObject.put("responseStatus", 200);
            } else if (request.getParameter("addt") != null && request.getParameter("id_tugas") != null && request.getParameter("tag") != null) {
                TugasDao tugasDao = new TugasDao();
                
                int idTugas = Integer.parseInt(request.getParameter("id_tugas"));
                String tag = request.getParameter("tag");
                
                tugasDao.addTag(idTugas, tag);
                tugasDao.updateTimestamp(idTugas);

                jObject.put("responseStatus", 200);
            } else if (request.getParameter("removet") != null && request.getParameter("id_tugas") != null && request.getParameter("tag") != null) {
                TugasDao tugasDao = new TugasDao();
                
                int idTugas = Integer.parseInt(request.getParameter("id_tugas"));
                String tag = request.getParameter("tag");
                
                tugasDao.removeTag(idTugas, tag);
                tugasDao.updateTimestamp(idTugas);

                jObject.put("responseStatus", 200);
            } else if (request.getParameter("suggestt") != null && request.getParameter("id_tugas") != null && request.getParameter("start") != null) {
                TugasDao tugasDao = new TugasDao();
                
                int idTugas = Integer.parseInt(request.getParameter("id_tugas"));
                String startWith = request.getParameter("start") + "%";
                
                jObject.put("suggestedTags", new JSONArray(tugasDao.getSuggestionTags(idTugas, startWith, 10)));
                jObject.put("responseStatus", 200);
            } else if (request.getParameter("removetugas") != null && request.getParameter("id_tugas") != null) {
                TugasDao tugasDao = new TugasDao();
                
                int idTugas = Integer.parseInt(request.getParameter("id_tugas"));
                
                tugasDao.deleteTugas(idTugas);

                jObject.put("responseStatus", 200);
            } else {
                jObject.put("responseStatus", 400);
                jObject.put("message", "Bad request");
            }
        } catch (Exception e) {
            jObject.put("responseStatus", 400);
            jObject.put("message", e.getMessage());
        }

        out.print(jObject);
    }
}
